superset-frontend/src/views/CRUD/data/dataset/DatasetList.tsx

```diff
@@ -118,6 +119,11 @@ type Dataset = {
   table_name: string;
 };
 
+interface VirtualDataset extends Dataset {
+  extra: any;
```

Comment:
nit: Describe `extra` (I think it's `Record<string, any>` but I'm not 100% sure)